app/services/credit_notes/generate_service.rb

```diff
@@ -12,9 +12,10 @@ def initialize(credit_note:, context: nil)
     def call
       return result.not_found_failure!(resource: "credit_note") if credit_note.blank? || !credit_note.finalized?
 
-      generate_pdf(credit_note) if should_generate_pdf?
-
-      SendWebhookJob.perform_later("credit_note.generated", credit_note)
+      if should_generate_pdf?
+        generate_pdf(credit_note)
+        SendWebhookJob.perform_later("credit_note.generated", credit_note)
```

Comment:
This webhook should be sent regardless of the `should_generate_pdf?` no?

Reply:
this webhook is really about the PDF generation. the only difference between `credit_note.generated` and `credit_note.created` is that the generated has the link to the PDF so I think we shouldn't send it.

Reply:
I'd say not send it as well, what's your thoughts @vincent-pochet @brunomiguelpinto ?

Reply:
ah, this is really a weird webhook name :D I forgot we also have `created`. Fine to skip in that case!